src/main/java/appeng/me/cache/PathGridCache.java

```diff
@@ -340,26 +345,24 @@ private void achievementPost()
 	{
 		if( this.lastChannels != this.getChannelsInUse() && AEConfig.instance().isFeatureEnabled( AEFeature.CHANNELS ) )
 		{
-			final Achievements currentBracket = this.getAchievementBracket( this.getChannelsInUse() );
-			final Achievements lastBracket = this.getAchievementBracket( this.lastChannels );
+			final IAdvancementTrigger currentBracket = this.getAchievementBracket( this.getChannelsInUse() );
+			final IAdvancementTrigger lastBracket = this.getAchievementBracket( this.lastChannels );
 			if( currentBracket != lastBracket && currentBracket != null )
 			{
-				final Set<Integer> players = new HashSet<>();
 				for( final IGridNode n : this.requireChannels )
 				{
-					players.add( n.getPlayerID() );
+					EntityPlayer player = AEApi.instance().registries().players().findPlayer( n.getPlayerID() );
+					if( player instanceof EntityPlayerMP )
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS134)